db/models.py

```diff
@@ -46,3 +53,73 @@ class MovieSession(models.Model):
 
     def __str__(self) -> str:
         return f"{self.movie.title} {str(self.show_time)}"
+
+
+class Order(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    user = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        on_delete=models.CASCADE
+    )
+
+    class Meta:
+        ordering = ["-created_at"]
+
+    def __str__(self) -> str:
+        return str(self.created_at)
+
+
+class Ticket(models.Model):
+    row = models.IntegerField()
+    seat = models.IntegerField()
+    movie_session = models.ForeignKey("MovieSession", on_delete=models.CASCADE)
+    order = models.ForeignKey("Order", on_delete=models.CASCADE)
+
+    class Meta:
+        constraints = [
+            UniqueConstraint(fields=["row", "seat", "movie_session"],
+                             name="unique_ticket_row_seat_movie_session")
+        ]
+
+    def __str__(self) -> str:
+        return (
+            f"{self.movie_session.movie.title}"
+            f" {self.movie_session.show_time}"
+            f" (row: {self.row}, seat: {self.seat})")
+
+    def clean(self) -> None:
+        cinema_hall_rows = self.movie_session.cinema_hall.rows
+        cinema_hall_seats = self.movie_session.cinema_hall.seats_in_row
+        if not 1 <= self.row <= cinema_hall_rows:
+            raise ValidationError(
+                {
+                    "row": [
+                        f"row number must be in available range: "
+                        f"(1, rows): (1, {cinema_hall_rows})"
+                    ]
+                }
+            )
+        if not 1 <= self.seat <= cinema_hall_seats:
+            raise ValidationError(
+                {
+                    "seat": [
+                        f"seat number must be in available range: "
+                        f"(1, seats_in_row): (1, {cinema_hall_seats})"
+                    ]
+                }
+            )
+
+    def save(
+            self,
+            force_insert: bool = False,
+            force_update: bool = False,
+            using: Any = None,
+            update_fields: Any = None) -> None:
+        self.full_clean()
+        return super(Ticket, self).save(
```

Comment:
do we need to pass `Ticket, self` into `super()`

Reply:
No, I do not need to pass Ticket, self into the super(). TY!